packages/components/rate/src/rate.vue

```diff
@@ -30,10 +30,10 @@
           ns.is('active', item <= currentValue),
         ]"
       >
-        <component
-          :is="iconComponents[item - 1]"
-          v-if="!showDecimalIcon(item)"
-        />
+        <template v-if="!showDecimalIcon(item)">
+          <component :is="activeComponent" v-show="item <= currentValue" />
```

Comment:
Is it more appropriate to use this code replacement? 🤔
```
<component :is="item <= currentValue ? activeComponent : voidComponent" />
```

Reply:
The original cause of the problem was caused by component switching, destruction and re-rendering.